src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc

```diff
@@ -440,6 +440,27 @@ class AresDnsResolverFactory : public ResolverFactory {
 
 }  // namespace grpc_core
 
+extern grpc_address_resolver_vtable* grpc_resolve_address_impl;
+static grpc_address_resolver_vtable* default_resolver;
+
+static void resolve_address_ares(const char* addr, const char* default_port,
+                                 grpc_pollset_set* interested_parties,
+                                 grpc_closure* on_done,
+                                 grpc_resolved_addresses** addrs) {
+  grpc_resolve_address_ares(addr, default_port, interested_parties, on_done,
+                            addrs);
+}
+
+static grpc_error* blocking_resolve_address_ares(
+    const char* name, const char* default_port,
+    grpc_resolved_addresses** addresses) {
+  return default_resolver->blocking_resolve_address(name, default_port,
```

Comment:
Why does this call the default resolver instead of an ares-specific function?

Reply:
The old mechanism for overriding the blocking resolver API relied on setting:
https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/resolve_address_posix.cc#L189

The ares resolver only ever used to override the async resolver API:
https://github.com/grpc/grpc/blob/master/src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc#L453

 This should not be a change in behavior.  AFAICT, the ares resolver always used the default blocking resolve address implementation.

Reply:
The reason that we did not previously override the blocking resolve address function is that the default blocking resolver address function simply calls the async version via the executor.

I think this relationship would be clearer if the DNS resolver API were in C++, because we could provide the blocking version in the base class and then simply not override it in the ares-specific subclass.